p2p/p2p-core/src/client/request_handler.rs

```diff
@@ -0,0 +1,144 @@
+use futures::TryFutureExt;
+use tower::ServiceExt;
+
+use cuprate_wire::{
+    admin::{
+        PingResponse, SupportFlagsResponse, TimedSyncRequest, TimedSyncResponse,
+        PING_OK_RESPONSE_STATUS_TEXT,
+    },
+    AdminRequestMessage, AdminResponseMessage, BasicNodeData,
+};
+
+use crate::{
+    client::PeerInformation,
+    constants::MAX_PEERS_IN_PEER_LIST_MESSAGE,
+    services::{
+        AddressBookRequest, AddressBookResponse, CoreSyncDataRequest, CoreSyncDataResponse,
+        PeerSyncRequest,
+    },
+    AddressBook, CoreSyncSvc, NetworkZone, PeerRequest, PeerResponse, PeerSyncSvc,
+    ProtocolRequestHandler,
+};
+
+#[derive(thiserror::Error, Debug, Copy, Clone, Eq, PartialEq)]
+enum PeerRequestHandlerError {
+    #[error("Received a handshake request during a connection.")]
+    ReceivedHandshakeDuringConnection,
+}
+
+/// The peer request handler, handles incoming [`PeerRequest`]s to our node.
+#[derive(Debug, Clone)]
+pub(crate) struct PeerRequestHandler<Z: NetworkZone, A, CS, PS, PR> {
```

Comment:
I added a request handler that can handle admin requests, this reduces the amount of requests that users have to handle.

Reply:
I also removed the generic peer request handler for a protocol request handler that only handles protocol requests, not admin.